### PR TITLE
Guide for project setup in windows (10 or higher)  issue #1175

### PR DESCRIPTION
**Pull request by 7saikat7**
Issue : [#1175](https://github.com/centerofci/mathesar/issues/1175)
I have updated the detailed process of setting up the project in a windows environment with best practices and also explained the errors one can face during the whole process.

### CODE REVIEW

File: community/setup_in_windows.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Pre-requisite
+- We assume you have Windows 10 or higher.
+- [Docker](https://docs.docker.com/get-docker/) is up and running properly in your system.
+
+# Solution 
+ - First search for "Turn Windows features on or off" , navigate to buttom and click the check-box named "Windows Subsystem For Linux".
+    
+ - Now you have to install any linux distro in your windows system .
+  Go to microsorft store and install any linux distro(Ubuntu prefered) and launch .
+ - If you do ls-la in ubuntu terminla you can see the linux files structure,
```
Comment (mathemancer) [inline community/setup_in_windows.md:18]:
```suggestion
 - If you do ls -la in ubuntu terminal you can see the linux files structure,
```

File: community/setup_in_windows.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     * so you have to access  window's files from ubuntu console, use this command to display all the storage drives avilable in windows , it will mount the storage .
+     ``
+     cd /mnt/
+     ``
+     * navigate to your desired file location e.g 
+     ```
+     cd /mnt/User/your_pc_name/Desktop/folder
+     ```
```
Comment (mathemancer) [inline community/setup_in_windows.md:26]: Why is this part needed? Is there a reason to access Windows' files for the purposes of the guide?

File: community/setup_in_windows.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ - Clone the repositories using command ``` git clone [repo-link](https://github.com/centerofci/mathesar)```
```
Comment (mathemancer) [inline community/setup_in_windows.md:27]: I think this would be better as a copy-pastable separate line rather than having a web link.

File: community/setup_in_windows.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    * Here you can ran into some problem regarding permission(chmod) in some .git-config files , it may occure as ubuntu is not configured properly in your windows machine just a quick restart can fix this issue .
```
Comment (mathemancer) [inline community/setup_in_windows.md:28]:
```suggestion
    * Here you may run into some problems regarding permissions in some .git-config files. It may occur as Ubuntu is not configured properly in your Windows machine. Restart your machine to fix this issue.
```

**Review by mathemancer — CHANGES_REQUESTED** [review]
Overall, pretty good. I can't test the instructions myself, since I don't use Windows. However, they do align with my understanding. Please address the formatting and grammar errors I noted.

File: community/setup_in_windows.md (hunk 2 above)
Comment (7saikat7) [reply community/setup_in_windows.md:26]: Yes sir it's needed, actually, we have to mount the storage so we can move to our desired directory(Eg.desktop) through ubuntu and then we have to clone the repo from the ubuntu console itself that's all , we have to do only once ! because rest of the time we can just open the folder in vscode from windows itself and work normally in windows  no need to open up ubuntu or any Linux distro

File: community/setup_in_windows.md (hunk 3 above)
Comment (7saikat7) [reply community/setup_in_windows.md:27]: ok sir !

File: community/setup_in_windows.md (hunk 4 above)
Comment (7saikat7) [reply community/setup_in_windows.md:28]: Thanks for your valuable suggestion, will create a pr soon!

File: community/setup_in_windows.md (hunk 2 above)
Comment (7saikat7) [reply community/setup_in_windows.md:26]: I will mention this in **Note:**  that it has to be done once, it seems a bit confusing for new users

Comment (mathemancer) [reply community/setup_in_windows.md:26]: Thank you for the explanation. That makes sense.

**Comment by 7saikat7** [comment]
Sorry for many no of commits, they can be done in a single commit . Please check if the process is understandable or not. 
Thank you!

File: community/setup_in_windows.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Project Set-Up In Windows Environment
+Windows users who want to run the Mathesar Docker development environment in WSL are advised to clone the repository in a Linux filesystem. 
+If you have tried running the project in windows you many ran into some issues including this
```
Comment (mathemancer) [inline community/setup_in_windows.md:3]:
```suggestion
If you have tried running the project in Windows you may ran into some issues including this 
```

File: community/setup_in_windows.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+# Project Set-Up In Windows Environment
+Windows users who want to run the Mathesar Docker development environment in WSL are advised to clone the repository in a Linux filesystem. 
+If you have tried running the project in windows you many ran into some issues including this 
+```
+service_1  | /usr/bin/env: ‘bash\r’: No such file or directory
+service_1  | 2022/03/30 16:20:21 Command exited with error: exit status 127
+```
+The complete guide is given below .
```
Comment (mathemancer) [inline community/setup_in_windows.md:8]:
```suggestion
The complete guide is given below.
```

File: community/setup_in_windows.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+## Pre-requisite
+- We assume you have Windows 10 or higher.
+- [Docker](https://docs.docker.com/get-docker/) is up and running properly in your system.
+
+# Solution 
+ - First search for "Turn Windows features on or off" , navigate to buttom and click the check-box named "Windows Subsystem For Linux".
```
Comment (mathemancer) [inline community/setup_in_windows.md:14]:
```suggestion
 - First search for "Turn Windows features on or off", navigate to the bottom, and click the check-box named "Windows Subsystem For Linux".
```

File: community/setup_in_windows.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    
+ - Now you have to install any linux distro in your windows system .
```
Comment (mathemancer) [inline community/setup_in_windows.md:16]:
```suggestion
 - Now you have to install a Linux distro in your Windows system .
```

File: community/setup_in_windows.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  Go to microsorft store and install any linux distro(Ubuntu prefered) and launch .
```
Comment (mathemancer) [inline community/setup_in_windows.md:17]:
```suggestion
  Go to the Microsoft store and install any Linux distro (Ubuntu preferred) and launch .
```

File: community/setup_in_windows.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+ - If you do ls -la in ubuntu terminal you can see the linux files structure,
```
Comment (mathemancer) [inline community/setup_in_windows.md:18]:
```suggestion
 - If you do `ls -la` in the Ubuntu terminal you can see the Linux file structure,
```

File: community/setup_in_windows.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+     * so you have to access  window's files from ubuntu console in order to reach your destination folder in windows, use the  command given below to display all the storage drives avilable in windows , it will mount the storage .
```
Comment (mathemancer) [inline community/setup_in_windows.md:19]: Please try a different wording. I'm not sure exactly what this means.

File: community/setup_in_windows.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+     ``
+     * navigate to your desired file location e.g 
+     ```
+     cd /mnt/User/your_pc_name/Desktop/folder
+     ```
+ - Clone the repository using command
+    ``` 
+    git clone https://github.com/centerofci/mathesar.git
+    ```
+    * Here you may run into some problems regarding permissions in some .git-config files. It may occur as Ubuntu is not configured properly in your Windows machine. Restart your machine to fix this issue.
```
Comment (mathemancer) [inline community/setup_in_windows.md:32]:
```suggestion
    * Here you may run into some problems regarding permissions in some `.git-config` files. It may occur as Ubuntu is not configured properly in your Windows machine. Restart your machine to fix this issue.
```

File: community/setup_in_windows.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+- Note: You have to perform these previous steps for the first time only ! after you have cloned the repo from ubuntu console in linux file sysyem then you can  simple close ubuntu and start working on the project as other django projects in windows.
```
Comment (mathemancer) [inline community/setup_in_windows.md:33]:
```suggestion
- Note: You have to perform these previous steps for the first time only! After you have cloned the repo from the Ubuntu console in the Linux file system then you can simply close Ubuntu and start working on the project as you would other Django projects in Windows.
```

File: community/setup_in_windows.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+- Now no Linux/ubuntu console is needed, just open the project in vs-code(or any ide you prefer) and follow along .
```
Comment (mathemancer) [inline community/setup_in_windows.md:34]:
```suggestion
- Now no Linux/Ubuntu console is needed; just open the project in vs-code(or any (IDE you prefer) and follow along.
```

File: community/setup_in_windows.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+ - Copy the `.env.example` file to `.env` like so:
+```
+cp .env.example .env
+```
+
+- From the repository's root directory, run this command (powershell prefered*):
```
Comment (mathemancer) [inline community/setup_in_windows.md:40]: This asterisk doesn't lead anywhere.

File: community/setup_in_windows.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+If it's your first time running the application, you'll also need to run database migrations and install Mathesar types and functions:
+```
+docker exec mathesar_service sh -c "python manage.py migrate && python install.py"
+```
+ You should now have a web server and database server running. Opening `http://localhost:8000` in your browser will open the application. For sample table data, you can create a new table in the UI using the `patents.csv` file found in `/mathesar/tests/data`. 
+
+It is recommended that you keep the Docker containers running while you make changes to the code. Any change to the code made locally will sync to the container and the version deployed at `http://localhost:8000` will always be the latest local version of the code.
+### Troubleshooting
+ * Running Script in powershell is disabled by default in windows , you have to change permission to run scripts  [Official Docs ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.security/set-executionpolicy?view=powershell-7.2)
```
Comment (mathemancer) [inline community/setup_in_windows.md:53]: I think the asterisk should point here, but github is interpreting this as a bullet-point.

**Review by mathemancer — CHANGES_REQUESTED** [review]
Hey, I took a closer look and found a number of typos. There's also a spot where I wasn't sure about what you meant.

File: community/setup_in_windows.md (hunk 11 above)
Comment (7saikat7) [reply community/setup_in_windows.md:19]: Should I add screenshot in different steps ...and also I found out a different process which is easier that this ...can I go for it ?

Comment (mathemancer) [reply community/setup_in_windows.md:19]: I'm not sure screenshots are the way. Maybe try the different process.

Comment (7saikat7) [reply community/setup_in_windows.md:19]: Sir, this suits best, other solution is solely vscode IDE oriented but this is a generalized solution.

File: community/setup_in_windows.md (hunk 16 above)
Comment (7saikat7) [reply community/setup_in_windows.md:53]: yes sir  , I have removed it .

**Review by mathemancer — APPROVED** [review]
Looks good with my edits.
